### ISSUE TITLE
Long delay before file watcher started working on big directory

### ISSUE THREAD

**Issue by garyng** [tilt-dev/tilt#5764]
### Expected Behavior
Maybe the file watchers should start working immediately after `tilt up`? Regardless of how many files the directory has.

### Current Behavior
Tilt seems to be taking a long time to setup the file watchers.

### Steps to Reproduce

`Tiltfile`:
```
local('date >> logs.txt')
```

Inside a directory with `12392` files, the file watcher works after <1s:
![image](https://user-images.githubusercontent.com/4453664/166609353-e88e1b03-7aa2-44f9-b147-2e4073842e8e.png)

Inside a directory with `183850` files, it took ~30min for the file watcher to start working:
![image](https://user-images.githubusercontent.com/4453664/166626233-3579e88b-a56b-4fc9-ae05-9caf40ba6322.png)

### Context

Setup:
- Tilt is running inside a docker container
- All the files in the container are mounted from a Windows host
- Docker socket is forwarded from Docker Desktop for Windows

#### `tilt doctor` Output

```
$ tilt doctor
Tilt: v0.28.1, built 2022-05-01
System: linux-amd64
---
Docker
- Host: unix:///var/run/docker.sock
- Server Version: 20.10.14
- API Version: 1.41
- Builder: 2
- Compose Version: v1.29.2 (build 5becea4c)
---
Kubernetes
- Env: docker-desktop
- Context: docker-desktop
- Cluster Name: docker-desktop
- Namespace: default
- Container Runtime: docker
- Version: v1.22.5
- Cluster Local Registry: none
---
Thanks for seeing the Tilt Doctor!
Please send the info above when filing bug reports. 💗

The info below helps us understand how you're using Tilt so we can improve,
but is not required to ask for help.
---
Analytics Settings
--> (These results reflect your personal opt in/out status and may be overridden by an `analytics_settings` call in your Tiltfile)
- User Mode: default
- Machine: 2e40bfee00900c30e704f0a120347b62
- Repo: 3oHS3Sdl6tpcRYbbMgRgBg==
```

**Comment by landism** [comment]
I've spent a bit of time trying to repro this without much luck. Just to confirm, is this the full content of your Tiltfile?
`local('date >> logs.txt')`

**Comment by garyng** [comment]
> I've spent a bit of time trying to repro this without much luck. Just to confirm, is this the full content of your Tiltfile? `local('date >> logs.txt')`

Yup! It's there just for logging, I was able to reproduce this even with an empty `Tiltfile`.
